ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -373,6 +373,55 @@ class SignalProducerSpec: QuickSpec {
 				expect(values).to(equal([ 1, 2, 3 ]))
 			}
 
+			it("should not deadlock in pair when started while sending") {
+				let (producer1, observer1) = SignalProducer<String, NoError>.buffer()
+				let (producer2, observer2) = SignalProducer<String, NoError>.buffer()
+
+				observer1.sendNext("A")
+				observer1.sendNext("B")
+				observer2.sendNext("1")
+				observer2.sendNext("2")
+
+				var valuePairs: [String] = []
+				producer1.startWithCompleted {
+					producer2.startWithCompleted {
+						valuePairs = []
+						producer1.startWithNext { value1 in
+							producer2.startWithNext { value2 in
+								valuePairs.append(value1 + value2)
+							}
+						}
+					}
+				}
+
+				observer1.sendCompleted()
+				observer2.sendCompleted()
+				expect(valuePairs).to(equal([ "A1", "A2", "B1", "B2" ]))
+			}
+
+			it("should send values in the order they were buffered") {
+				let (producer, observer) = SignalProducer<Int, NoError>.buffer()
+
+				observer.sendNext(1)
+				observer.sendNext(2)
+
+				var values: [Int] = []
+				producer.startWithNext { value in
+					values.append(value)
+					if value == 1 {
+						// the expected behavior is for the other queue to block until
+						// the first queue wakes up and receives all the buffered events
+						dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
+							observer.sendNext(3)
+						}
+						usleep(1000)
```

Comment:
I need to test that another thread will wait until the original one replays all the buffer to the observer. It doesn't seem like `TestScheduler` can do this. Since it doesn't use any other threads, it will probably just deadlock.